### PR TITLE
Add Florence2 model support and tests

### PR DESCRIPTION
**Pull request by fjaijf**
### PR 类型

新增模型支持 / Models

### PR 变更

新增 Florence2 在 PaddleFormers 中的模型接入，包括配置、模型、Processor、ImageProcessor、Auto 注册、SFT template、单测与验证脚本。

### 主要内容

1. 新增 `paddleformers/transformers/florence2`
   - `Florence2Config`
   - `Florence2ForConditionalGeneration`
   - `Florence2VisionModel`
   - `Florence2Processor`
   - `Florence2ImageProcessor`

2. 接入 Auto 入口
   - `AutoConfig`
   - `AutoModel`
   - `AutoProcessor`
   - `AutoImageProcessor`

3. 支持从原始 HuggingFace 权重加载
   - 本地验证模型路径：`/sda/housaijie/code/Florence-2-base`
   - 支持 `convert_from_hf=True`
   - 已验证 `model.safetensors` 权重可完整加载

4. 新增测试
   - `tests/transformers/florence2/test_modeling.py`
   - `tests/transformers/florence2/test_processor.py`

#### 1. 单卡前向精度对齐

使用本地真实 Florence2-base 权重，对齐 transformers/PyTorch 与 PaddleFormers 的 FP32 forward logits。

```text
logits shape: (1, 1, 51289)
logits mean abs diff: 0.00010860827751457691
logits p95 abs diff: 0.00026438236236572257
logits p99 abs diff: 0.0003495216369628906
logits max abs diff: 0.0006575584411621094

满足 1e-2 量级要求。

2. 生成结果对齐
使用 greedy search，固定输入 <CAPTION> + 测试图片，对比前 10 个生成 token。

torch tokens:
[0, 102, 2440, 3618, 19, 10, 1104, 1424, 2, 2]

paddle tokens:
[0, 102, 2440, 3618, 19, 10, 1104, 1424, 2, 2]

first10_equal: True

3. 单测结果
执行命令：

> [!NOTE]
>  miniconda3/envs/paddleformers/bin/python -m unittest \
  tests.transformers.florence2.test_modeling \
  tests.transformers.florence2.test_processor

### CODE REVIEW

File: paddleformers/transformers/florence2/modeling.py [hunk 1]
```diff
@@ -0,0 +1,977 @@
...
+        output_attentions=False,
+        output_hidden_states=False,
+        return_dict=True,
+        **kwargs,
+    ):
+        if labels is not None and input_ids is not None and labels.shape == input_ids.shape:
+            input_ids, labels, attention_mask = self._split_sft_inputs(input_ids, labels, attention_mask)
+        image_features = None
+        if encoder_output is None and encoder_outputs is None and inputs_embeds is None:
+            inputs_embeds = self.get_input_embeddings()(input_ids)
```
Comment (risemeup1111) [inline paddleformers/transformers/florence2/modeling.py]: P1 这里会在处理图像前无条件对 `input_ids` 做 embedding；因此 `forward(pixel_values=..., decoder_input_ids=...)` 或 `generate(pixel_values=...)` 传入 `input_ids=None` 时会直接失败。Florence-2 的图像条件调用允许没有文本 embedding（合并逻辑应直接使用 `image_features`），而本 PR 的 `test_generate_without_input_ids` 也被留成了空测试。请仅在 `input_ids` 非空时查 embedding，并让合并函数支持 `inputs_embeds=None`，同时恢复该路径的测试。

File: paddleformers/transformers/florence2/modeling.py [hunk 2]
```diff
@@ -0,0 +1,977 @@
...
+                logits.reshape([-1, self.config.vocab_size]),
+                labels.reshape([-1]),
+                ignore_index=-100,
+            )
+        if not return_dict:
+            return ((loss, logits) if loss is not None else (logits,)) + outputs[1:]
+        return Seq2SeqLMOutput(
+            loss=loss,
+            logits=logits,
+            past_key_values=outputs.past_key_values,
```
Comment (risemeup1111) [inline paddleformers/transformers/florence2/modeling.py:749]: P1 这里返回的是 `Seq2SeqLMOutput`，但当前 `GenerationMixin.update_model_kwargs_for_generation` 只会从 tuple 或 `CausalLMOutputWithPast` 写回 `past_key_values`。因此这里生成的缓存不会进入下一步，`prepare_inputs_for_generation` 每次都收到 `None`，默认生成会反复重算完整 decoder 前缀。请为 Seq2Seq 输出接入缓存更新（在模型中覆写或扩展 mixin 均可），并增加第二步缓存非空及 cached/non-cached parity 测试。

File: tests/transformers/florence2/test_modeling.py [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+        self.assertEqual(decoder_labels.tolist(), [[20, 21, 2]])
+        self.assertEqual(source_mask.tolist(), [[1, 1, 1]])
+
+
+class Florence2ModelIntegrationTest(ModelTesterPretrainedMixin, unittest.TestCase):
+    base_model_class = Florence2ForConditionalGeneration
+    hf_remote_test_model_path = None
+    paddlehub_remote_test_model_path = None
+
+    @unittest.skip("Florence2 tiny pretrained checkpoint is not available yet.")
```
Comment (risemeup1111) [inline tests/transformers/florence2/test_modeling.py]: P1 这里及后续所有 pretrained 测试都被跳过，而本地 save/load 测试只验证随机 Paddle 权重的自洽，完全没有覆盖本 PR 的 HF checkpoint 转换、权重转置和真实 Florence-2 输出。对于这次完整模型端口，这是必需的集成验证；现有 shape 测试无法发现转换后数值偏差。请至少加入可执行的 HF→Paddle 加载与固定输入 logits/generation parity（可以标记为 slow）。

File: paddleformers/transformers/florence2/processor.py [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+                    labels.append(phrase.strip())
+            return {task: {"polygons": polygons, "labels": labels}}
+        if task == "<REGION_PROPOSAL>":
+            values = [int(value) for value in re.findall(r"<loc_(\d+)>", clean_text)]
+            bboxes = [
+                self._dequantize(values[index : index + 4], image_size) for index in range(0, len(values) - 3, 4)
+            ]
+            return {task: {"bboxes": bboxes, "labels": [""] * len(bboxes)}}
+
+        phrase_pattern = r"([^<]+)((?:<loc_\d+>){4,})"
```
Comment (risemeup1111) [inline paddleformers/transformers/florence2/processor.py]: P2 `task_prompts_with_input` 已公开 `<OPEN_VOCABULARY_DETECTION>`，其输出既可能是 bbox 也可能是 polygon；但这里统一落入 bbox 正则。遇到 `cat<poly><loc_...>...</poly>` 时只会把前四个坐标误报成 bbox，其余 polygon 坐标被丢弃。请为该 task 增加 bbox/polygon 分流并保持对应 labels，测试两种输出格式。

**Review by risemeup1111 — CHANGES_REQUESTED** [review]
| 序号 | 位置 | 优先级 | 状态 |
| --- | --- | --- | --- |
| 1 | [纯图像调用](https://github.com/PaddlePaddle/PaddleFormers/pull/4869#discussion_r3756741450) | P1 | ✅ |
| 2 | [生成缓存传递](https://github.com/PaddlePaddle/PaddleFormers/pull/4869#discussion_r3756743882) | P1 | ✅ |
| 3 | [预训练权重集成验证](https://github.com/PaddlePaddle/PaddleFormers/pull/4869#discussion_r3756745628) | P1 | ✅ |
| 4 | [开放词表 polygon 后处理](https://github.com/PaddlePaddle/PaddleFormers/pull/4869#discussion_r3756748169) | P2 | ✅ |

<div align="right">
   <sup>Powered by Nyanpasu with gpt-5.5 xhigh, please check the suggestions carefully.</sup>
</div>

File: paddleformers/transformers/florence2/modeling.py (hunk 1 above)
Comment (risemeup1111) [reply paddleformers/transformers/florence2/modeling.py]: 已修复：现在仅在 `input_ids` 非空时创建文本 embedding，纯图像的 `forward/generate` 会直接使用图像特征，并补充了对应测试。

File: paddleformers/transformers/florence2/processor.py (hunk 4 above)
Comment (risemeup1111) [reply paddleformers/transformers/florence2/processor.py]: 已修复：`<OPEN_VOCABULARY_DETECTION>` 现在区分 bbox 与 polygon 输出，并新增两种格式的后处理测试。

File: paddleformers/transformers/florence2/modeling.py (hunk 2 above)
Comment (risemeup1111) [reply paddleformers/transformers/florence2/modeling.py:749]: 已修复：模型现在会把 `Seq2SeqLMOutput.past_key_values` 写回生成参数，并补充了缓存更新及 cached/non-cached logits 对齐测试。

**Review by risemeup1111 — APPROVED** [review]
已复查新增提交：纯图像 forward/generate、Seq2Seq generation cache、HF checkpoint parity 以及开放词表 polygon 后处理均已补齐对应实现与测试。当前未发现未解决的代码问题。
